File: tester/harmony/rnoh/src/main/ets/RNOHCorePackage/turboModules/ToastAndroidTurboModule.ts
import type { TurboModuleContext } from '../../RNOH/TurboModule';
import { TurboModule } from '../../RNOH/TurboModule';
import promptAction from '@ohos.promptAction';
import { BusinessError } from '@ohos.base';
import { RNOHLogger } from '../../RNOH/RNOHLogger';

const ToastAndroidConstants = {
  SHORT: 2000,
  LONG: 3500,
  TOP: 0,
  BOTTOM: 1,
  CENTER: 2,
}

export class ToastAndroidTurboModule extends TurboModule {
  public static readonly NAME = 'ToastAndroid';
  private logger: RNOHLogger;

  constructor(protected ctx: TurboModuleContext) {
    super(ctx);
    this.logger = this.ctx.logger.clone("ToastAndroidTurboModule")

  }

  getConstants() {
    return ToastAndroidConstants;
  }

  show(message: string, duration: number) {
    try {
      promptAction.showToast({
        message,
        duration,
      })
    } catch (error) {
      const errorMessage = (error as BusinessError).message;
      const errorCode = (error as BusinessError).code;
      this.logger.error(`Show toast failed with code ${errorCode} and message ${errorMessage}`)
    }
  }

  showWithGravity(message: string, duration: number, _gravity: number) {
    this.show(message, duration);
  }

  showWithGravityAndOffset(message: string, duration: number, _gravity: number, _xOffset: number, _yOffset: number) {
    this.show(message, duration);
  }
}

File: tester/harmony/react_native_openharmony/src/main/cpp/RNOHCorePackage/TurboModules/ToastAndroidTurboModule.cpp
#include "ToastAndroidTurboModule.h"
#include "RNOH/ArkTSTurboModule.h"

using namespace facebook;

namespace rnoh {
ToastAndroidTurboModule::ToastAndroidTurboModule(
    const ArkTSTurboModule::Context ctx,
    const std::string name)
    : ArkTSTurboModule(ctx, name) {
  methodMap_ = {
      ARK_METHOD_METADATA(getConstants, 0),
      ARK_METHOD_METADATA(show, 2),
      ARK_METHOD_METADATA(showWithGravity, 3),
      ARK_METHOD_METADATA(showWithGravityAndOffset, 5),
  };
}
} // namespace rnoh
